README.md

```diff
@@ -320,6 +320,56 @@ describe('MockRender', () => {
 });
 ```
 
+## MockProvider
+
+Provides simple way to providers/services when including them in the testing module. 
+
+By default provider will be mocked with an empty method, but if you provide factory method for mocking, all methods will be mocked with that setting. This is useful for example when we want to mock service methods with something like `jasmine.createSpy` or `jest.fn()`.
+
+### Usage Example
+```typescript
+describe('MockProvider', () => {
+
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      providers: [
+        MockProvider(RealService)
+      ]
+    });
+
+    const service = TestBed.get(RealService);
+  });
+});
+```
+
+With factory method:
+```typescript
+describe('MockProvider', () => {
+
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      providers: [
+        MockProvider(RealService, (methodName) => jasmine.createSpy(methodName))
+      ]
+    });
+  });
+
+  it('call service', () => {
+    const service = TestBed.get(RealService);
+
+    // Assuming that component calls the service:
+    expect(service.callMethod).toHaveBeenCalled();
+  });
+});
+```
+
+### Limitations
+
+Current limitations of MockProvider are not being able to mock:
+- properties
```

Comment:
Actually this would be really useful. We use [spectator](https://github.com/NetanelBasal/spectator) which has `mockProvider` and I added support for mocking properties here https://github.com/NetanelBasal/spectator/pull/58

it works like this:

```
mockProvider(ActivatedRoute, {
    url: activatedRouteUrl,
    snapshot: {}
})
```

`url` and `snapshot` are properties defined on the mock. Other methods are automatically mocked with `jasmine.Spy`.

Reply:
I couldn't find a way to read properties from class definition on runtime. But adding a list of them, could be done, indeed.